[CI] NetworkServiceTests.testPublishAnyLocalV4 failed on Java 14

`NetworkServiceTests.testPublishAnyLocalV4` failed in https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA=openjdk12,ES_RUNTIME_JAVA=openjdk14,nodes=general-purpose/370/console

It is interesting that the runtime Java here is Java 14, which is not yet released, "Runtime JDK Version   : 13 (Oracle Corporation 14-ea [OpenJDK 64-Bit Server VM 14-ea+25-1178])" to be precise.

The error was:

```
java.io.IOException: failed to check if interface [veth854822e] is up
Caused by: java.net.SocketException: No such device (getFlags() failed)
at java.net.NetworkInterface.isUp0(Native Method)
at java.net.NetworkInterface.isUp(NetworkInterface.java:458)
at org.elasticsearch.common.network.NetworkUtils.isUp(NetworkUtils.java:173)
at org.elasticsearch.common.network.NetworkUtils.filterAllAddresses(NetworkUtils.java:160)
at org.elasticsearch.common.network.NetworkUtils.getAllAddresses(NetworkUtils.java:201)
at org.elasticsearch.common.network.NetworkService.resolvePublishHostAddresses(NetworkService.java:155)
at org.elasticsearch.common.network.NetworkServiceTests.testPublishAnyLocalV4(NetworkServiceTests.java:108)
at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:564)
at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
at java.lang.Thread.run(Thread.java:832)
```

The REPRO command is:

```
./gradlew ':server:test' --tests "org.elasticsearch.common.network.NetworkServiceTests.testPublishAnyLocalV4" \
  -Dtests.seed=AA7FF538B7DFAC9D \
  -Dtests.security.manager=true \
  -Dtests.locale=ms-MY \
  -Dtests.timezone=Europe/Skopje \
  -Dcompiler.java=12
```

This doesn't reproduce using Java 12 as the runtime Java, so I am guessing it's related to Java 14.  I'm not sure if this is something that needs to be fixed in the networking code, the Java 14 code or our CI worker setup, so pinging both teams initially for awareness.

--- a/server/src/main/java/org/elasticsearch/common/network/NetworkUtils.java
+++ b/server/src/main/java/org/elasticsearch/common/network/NetworkUtils.java
@@ -172,6 +172,10 @@ public abstract class NetworkUtils {
         try {
             return intf.isUp();
         } catch (final SocketException e) {
+            // virtual ethernet devices come and go, we will treat such a device that disappeared as not being up
+            if (intf.getName().startsWith("veth") && e.getMessage().equals("No such device (getFlags() failed)")) {
+                return false;
+            }
             throw new IOException("failed to check if interface [" + intf.getName() + "] is up", e);
         }
     }

